src/backend/function/Fusion.Resources.Functions/Functions/ProfileSync.cs

```diff
@@ -70,43 +71,40 @@ public async Task SyncProfiles([TimerTrigger("0 0 5 * * *", RunOnStartup = false
         {
             log.LogInformation("Synchronizing external person personnel in Resources API");
 
-            var personnel = await GetAllExternalPersonnelAsync();
+            var personnel = await GetAllActiveExternalPersonnelAsync();
 
-            var mailsToEnsure = personnel!
+            var usersToEnsure = personnel!
                 .Where(p => !string.IsNullOrWhiteSpace(p.Mail))
-                .Select(p => p.Mail)
+                .Select(usr => new ExternalPersonnelId(usr.AzureUniquePersonId, usr.Mail))
                 .Distinct()
                 .ToList();
 
-            var mailsToEnsureCount = mailsToEnsure.Count;
-            if (mailsToEnsureCount == 0)
+            if (usersToEnsure.Count == 0)
             {
-                log.LogInformation($"Found {mailsToEnsureCount} profiles to ensure. Aborting...");
+                log.LogInformation($"Found no profiles to ensure. Aborting...");
                 return;
             }
 
-            var ensureResult = await EnsurePersonsAsync(mailsToEnsure);
+            // OriginalIdentifier contains AzureUniqueId if found, or Mail.  
+            var ensureResult = await EnsurePersonsAsync(usersToEnsure.Select(x => x.OriginalIdentifier).ToList());
 
+            // log ensure results
             var successCount = ensureResult.Count(x => x.Success);
             var failureCount = ensureResult.Count(x => x.Success == false);
-            if (successCount > 0)
-            {
-                log.LogInformation($"Successfully ensured {successCount} profiles");
-            }
-
-            if (failureCount > 0)
-            {
-                log.LogWarning($"Failed to ensure {failureCount} profiles");
-            }
+            if (successCount > 0) log.LogInformation($"Successfully ensured {successCount} profiles");
+            if (failureCount > 0) log.LogWarning($"Failed to ensure {failureCount} profiles");
 
             foreach (var person in ensureResult)
             {
-                var resourcesPerson = personnel.First(p => p.Mail == person.Identifier);
+                // person.Identifier contains AzureUniqueId if found, or Mail.   
+                var personIdentifier = new ExternalPersonnelId(person.Identifier!);
+                var resourcesPerson = personIdentifier.Type == ExternalPersonnelId.IdentifierType.UniqueId ? personnel.First(p => p.AzureUniquePersonId == personIdentifier.UniqueId) : personnel.First(p => p.Mail == personIdentifier.Mail);
```

Comment:
Probably not a big deal, but this is O(n^2). Could we store the result from `GetAllActiveExternalPersonnelAsync();` in a dictionary indexed by fusion identifer?